examples/local-explorer/explorer-watcher.mjs

```diff
@@ -0,0 +1,92 @@
+import { watchFile } from "fs";
+import { readFile } from "fs/promises";
+import { spawn } from "child_process";
+import { fileURLToPath } from "url";
+import process from "process";
+import path from "path";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const EXPLORER_DIR = path.resolve(__dirname, "..", "..", "packages", "explorer");
+const WORLDS_JSON_PATH = path.resolve(__dirname, "packages", "contracts", "worlds.json");
+
+const MODE = process.env.MODE || "development";
+const CHAIN_ID = parseInt(process.env.CHAIN_ID || 31337);
+let explorerProcess = null;
+
+async function startExplorer() {
+  process.env.INIT_PWD = process.cwd();
+  process.env.NEXT_PUBLIC_CHAIN_ID = CHAIN_ID;
```

Comment:
mutating the env feels dangerous, especially when we can just pass this into `spawn` via `env` arg

Reply:
ah missed it, very good point, now passing environment variables as `spawn` args,